lib/screens/post_slot_screen.dart

```diff
@@ -0,0 +1,219 @@
+import 'package:flutter/material.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:uuid/uuid.dart';
+
+class PostSlotScreen extends StatefulWidget {
+  @override
+  _PostSlotScreenState createState() => _PostSlotScreenState();
+}
+
+class _PostSlotScreenState extends State<PostSlotScreen> {
+  String uuid = Uuid().v1();
+  Position position;
+  Position selectedPosition;
+  String selectedLocation;
+  String slotImage;
+  List<Marker> markers = [];
+  User user;
+
+  // Controllers
+  TextEditingController titleController = TextEditingController();
+  TextEditingController descriptionController = TextEditingController();
+  TextEditingController dailyController = TextEditingController();
+  TextEditingController hourlyController = TextEditingController();
+
+  initState() {
+    super.initState();
+    user = FirebaseAuth.instance.currentUser;
+    selectedPosition = position;
+    selectedLocation = 'Tap on map to select location...';
+    getPosition();
+  }
+
+  getPosition() async {
+    Position position = await Geolocator()
+        .getCurrentPosition(desiredAccuracy: LocationAccuracy.high);
+    if (position != null) {
+      setState(() {
+        this.position = position;
+      });
+    }
+  }
+
+  setMarker() {
+    setState(() {
+      markers = [];
+      markers.add(
+        Marker(
+          position:
+              LatLng(selectedPosition.latitude, selectedPosition.longitude),
+          markerId: MarkerId('CurrentSelectedLatLng'),
+        ),
+      );
+      getLocation();
+    });
+  }
+
+  getLocation() async {
+    List<Placemark> placemarks = await Geolocator().placemarkFromCoordinates(
+        selectedPosition.latitude, selectedPosition.longitude);
+    setState(() {
+      selectedLocation = placemarks[0].thoroughfare +
+          ', ' +
+          placemarks[0].postalCode +
+          ', ' +
+          placemarks[0].locality;
+    });
+  }
+
+  postSlot() {
+    // TODO: Add validation
+    CollectionReference slotsDb =
+        FirebaseFirestore.instance.collection('slots');
+    slotsDb.doc(uuid).set({
+      'title': titleController.text,
+      'latitude': selectedPosition.latitude,
+      'longitude': selectedPosition.longitude,
+      'daily': dailyController.text,
+      'hourly': hourlyController.text,
+      'imageUrl': slotImage,
+      'userUid': user.uid,
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    StorageReference imgStorage = FirebaseStorage.instance.ref();
+    if (position == null) {
+      return Material(child: Center(child: Text("loading...")));
+    } else {
+      return Scaffold(
+        appBar: AppBar(
+          centerTitle: true,
+          title: Text('Add new slot'),
+        ),
+        body: Column(
+          children: <Widget>[
+            MediaQuery.of(context).viewInsets.bottom > 1
+                ? Container()
```

Comment:
What is the use of an empty container?

Instead, isn't it possible to skip this child at all?

Reply:
Unfortunately there is no way to skip it, or at least I wasn't able to found a solution better than that.
If I would return null or use the ?? operator, there will be an error.